apps/mobile/src/screens/signup/compose/signup-form.tsx

```diff
@@ -0,0 +1,77 @@
+import { joiResolver } from '@hookform/resolvers/joi';
+import { useNavigation } from '@react-navigation/native';
+import Joi from 'joi';
+import { Button, FormControl, useToast } from 'native-base';
+import { FormProvider, useForm } from 'react-hook-form';
+import { KeyboardAvoidingView, Platform } from 'react-native';
+
+import { FormField } from './form-field';
+import { UserType } from '../../../../../../shared/types';
+import { useSignUp } from '../../../hooks/use-onboarding-provider';
+import Routes from '../../../routes';
+
+type RegisterUserFormProps = {
+  defaultValues?: Partial<UserType>;
+};
+
+const createUserSchema = Joi.object({
+  name: Joi.string().required(),
+  email: Joi.string()
+    .email({ tlds: { allow: false } })
+    .required(),
+  password: Joi.string().min(6).required(),
+});
+
+export const SignUpForm = ({ defaultValues }: RegisterUserFormProps) => {
+  const saveUserForm = useForm<Partial<UserType>>({
+    resolver: joiResolver(createUserSchema),
+    mode: 'onChange',
+    defaultValues,
+  });
+
+  const toast = useToast();
+
+  const { navigate } = useNavigation();
+  const { updateUserInfo } = useSignUp();
+
+  const onConfirm = async (data: UserType) => {
+    const isValid = await saveUserForm.trigger();
+
+    if (!isValid) {
+      toast.show({
+        description: `Invalid data`,
+      });
+
+      return;
+    }
+
+    updateUserInfo(data);
+
+    navigate(Routes.GetLocation as never);
```

Comment:
So as mentioned earlier, you could get rid of the onboarding context and pass the user data filled up in the form through the routes parameters here.